### PR TITLE
Add SizedBox.fromSize() constructor

### PR DESCRIPTION
**Pull request by HansMuller**
See https://github.com/flutter/flutter/pull/8112#discussion_r100924808

### CODE REVIEW

File: packages/flutter/lib/src/widgets/basic.dart [hunk 1]
```diff
@@ -829,6 +829,11 @@ class SizedBox extends SingleChildRenderObjectWidget {
       height = double.INFINITY,
       super(key: key, child: child);
 
+  SizedBox.fromSize({ Key key, Widget child, Size size })
+    : width = size?.width,
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/basic.dart]: can this be a const constructor?

Comment (Hixie) [inline packages/flutter/lib/src/widgets/basic.dart]: we should probably add dartdocs

Comment (HansMuller) [inline packages/flutter/lib/src/widgets/basic.dart]: I was unable to figure out a way to make it one. Even if you assume size is non-null, initializing the fields to size.width and size.height produces a 'not a const expression' error.

**Comment by Hixie** [comment]
LGTM

**Pull request merged**